fsspec/utils.py

```diff
@@ -302,3 +302,32 @@ def stringify_path(filepath):
     elif isinstance(filepath, pathlib.Path):
         return str(filepath)
     return filepath
+
+
+def inherit_docstrings(cls):
+    from .spec import AbstractFileSystem
+
+    for method in dir(cls):
+        if method.startswith("_"):
+            continue
+
+        m = getattr(cls, method)
+        n = getattr(AbstractFileSystem, method, None).__doc__
+        if not callable(m) or not n or n in (m.__doc__ or ""):
+            # ignore if a) not a method, b) no superclass doc
+            # c) already includes docstring
+            continue
+        if m.__doc__ and hasattr(getattr(AbstractFileSystem, method), "__doc__"):
+            template = """\
```

Comment:
Just a heads up, I tried this out with `s3fs` and am getting a `numpydoc` error from having multiple `Parameter` sections

<details>
<summary>Traceback:</summary>

```terminal
sphinx-build -b html -d build/doctrees   source build/html
Running Sphinx v2.2.1
making output directory... done
WARNING: html_static_path entry '_static' does not exist
building [mo]: targets for 0 po files that are out of date
building [html]: targets for 4 source files that are out of date
updating environment: [new config] 4 added, 0 changed, 0 removed
reading sources... [ 25%] api
Exception occurred:
  File "/Users/jbourbeau/miniconda/envs/s3fs-dev/lib/python3.7/site-packages/numpydoc/docscrape.py", line 422, in _error_location
    raise ValueError(msg)
ValueError: The section Parameters appears twice in the docstring of <function S3FileSystem.ls at 0x10a7e6378> in /Users/jbourbeau/github/dask/s3fs/s3fs/core.py.
The full traceback has been saved in /var/folders/m3/6sdjxnm164q0bf5pw5dj24t80000gp/T/sphinx-err-n0jao0x7.log, if you want to report the issue to the developers.
Please also report this if it was a user error, so that a better error message can be provided next time.
A bug report can be filed in the tracker at <https://github.com/sphinx-doc/sphinx/issues>. Thanks!
make: *** [html] Error 2
```

</details>

What if we don't include the upstream docstring if it's defined for the subclass?

Reply:
Thanks for trying that out. Older versions of numpydoc may be OK with it (that's what we do on dask/dask)

FWIW, that's my preference. I'd rather downstream packages explicitly reference the upstream docs. Something like

```python
def copy(...):
    """
    My docstring.

    See Also
    ---------
    fsspec.AbstractFileSystem.copy
```

will produce a link to the fsspec docs in the HTML version, as long as intersphinx config is setup.

Reply:
I think in dask/dask if the subclass defines it's own `Parameters` section, then the corresponding upstream section has it's title modified to be `Extra Parameters`. 

Your linking to `fsspec` docs proposal seems like a good way to go

Reply:
Does OK by you @martindurant? If so, I'll close this PR and make PRs in downstream libraries to update their docs.

Reply:
Yes, that's OK. We can even reuse the function from dask.

Reply:
Sorry, may not have been clear. My preference (and I think @jrbourbeau's) is to no implicitly inherit the parent docstrings. Rather, the children should either

1. Link to the parent docstrings when they overwite
2. Do the magic to combine docstrings if they so desire.